File: hsg9-tinh/2023-2024-khanhhoa/tudai/tudai.cpp
#include <iostream>
#include <sstream>

#define input_file "tudai.inp"
#define output_file "tudai.out"

using namespace std;

string line;
string longest_word;
int max_length = 0;

void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    freopen(input_file, "r", stdin);

    getline(cin, line);
}


void process()
{
    // Khởi tạo stringstream từ chuỗi line
    stringstream ss(line);

    string word;

    // Đọc từng từ trong ss vào word
    while (ss >> word)
    {
        int len = word.length();
        if (len >= max_length)
        {
            max_length = len;
            longest_word = word;
        }
    }
}


void output()
{
    freopen(output_file, "w", stdout);

    cout << max_length << '\n' << longest_word;
}


int main()
{
    input();
    process();
    output();

    return 0;
}

File: hsg9-tinh/2023-2024-khanhhoa/tudai/tudai.py
import os
import sys

input_file = os.path.join(sys.path[0], 'tudai.inp')
output_file = os.path.join(sys.path[0], 'tudai.out')

line = ''
longest_word = ''
max_length = 0

def input():
    global line

    with open(input_file, 'r') as f:
        line = f.readline().strip()


def process():
    global line, longest_word, max_length

    # Tách chuỗi line thành list gồm các từ
    words = line.split()

    # Duyệt từng từ trong list words
    for word in words:
        length = len(word)

        if length >= max_length:
            max_length = length
            longest_word = word


def output():
    global longest_word, max_length

    with open(output_file, 'w') as fo:
        fo.write(f'{max_length}\n{longest_word}')


if __name__ == '__main__':
    input()
    process()
    output()